components/ModFacilitySearchbar.vue

```diff
@@ -0,0 +1,181 @@
+                "
+                :class="computedDropdownClass"
+            >
+                <!-- Search limited to 10 items -->
+                <li
+                    v-for="(facility, index) in filteredFacilities.slice(0, 10)"
+                    :key="facility.id"
+                    class="flex justify-between divide-x cursor-pointer"
+                    :class="[
+                        selectedFacilities.has(facility) && 'bg-primary-text-muted',
+                        selectedFacilityIndex === index ? 'bg-primary-hover text-primary-inverted' : 'opacity-95',
+                    ]"
+                    @click="(event) => { event.preventDefault(); handleListItemClick(facility) }"
+                    @mousedown="(event) => {
+                        // Prevent search input from being blurred
+                        event.preventDefault()
+                    }"
+                    @mousemove="selectedFacilityIndex = index"
+                >
+                    <div class="flex flex-col overflow-x-auto whitespace-nowrap m-3">
+                        <span class="text-xs">{{ facility.id }}</span>
+                        <span class="">{{ `${facility.nameEn} / ${facility.nameJa} ` }}</span>
+                    </div>
+                    <div class="flex items-center">
+                        <SVGCheckMark
+                            class="h-4 m-3"
+                            :class="[
+                                selectedFacilities.has(facility) ? 'opacity-100' : 'opacity-10',
+                            ]"
+                        />
+                    </div>
+                </li>
+                <li
+                    v-if="!filteredFacilities.length"
+                    class="m-3 cursor-default"
+                >
+                    <!-- TODO: Change i18n key if necessary -->
+                    <span>{{ $t('modFacilitySearchbar.noFacilitiesWereFound') }}</span>
+                </li>
+            </ul>
+        </div>
+    </div>
+</template>
+
+<script lang="ts" setup>
+import { computed, onMounted, onUnmounted, ref, watch } from 'vue'
+import SVGCheckMark from '~/assets/icons/check-mark.svg'
+import SVGLookingGlass from '~/assets/icons/looking-glass.svg'
+import { useFacilitiesStore } from '#build/imports'
+import { useFacilitySearchbarStore } from '~/stores/facilitySearchbarStore'
+import type { Facility } from '~/typedefs/gqlTypes'
+
+//The limit for fetching facilities is 400 at the current date (30/11/2024)
+const facilitiesStore = useFacilitiesStore()
+const facilitySearchbarStore = useFacilitySearchbarStore()
+
+const searchInput = ref('')
+const searchInputElement = ref<HTMLInputElement | null>(null)
+const filteredFacilities = ref<Facility[]>([])
+const selectedFacilities = ref(facilitySearchbarStore.selectedFacilities)
+const selectedFacilityIndex = ref(0)
+
+const handleListItemClick = (facility: Facility) => {
+    searchInputElement.value?.focus()
```

Comment:
let's add a comment here to explain why were focusing here. To that point, I'm not sure why it's focused 🤔

Reply:
This is a leftover. Removing it! 😊